ironfish/src/mining/stratum/stratumPeers.ts

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import net from 'net'
+import { Event } from '../../event'
+import { Config } from '../../fileStores/config'
+import { createRootLogger, Logger } from '../../logger'
+import { SetTimeoutToken } from '../../utils'
+import { StratumServerClient } from './stratumServerClient'
+
+const TWO_HOURS_MS = 2 * 60 * 60 * 1000
+const PEERS_TICK_MS = 10000
+
+export class StratumPeers {
+  readonly logger: Logger
+  readonly maxConnectionsByIp: number
+
+  readonly onBanned = new Event<[StratumServerClient]>()
+
+  banCount = 0
```

Comment:
This is very much a preference nit, so up to you, but I think a getter is probably a little less prone to bugs. If we prune the expired bans, we can just do a real simple `this.bannedByIp.size`

Reply:
This a count of bans, not people banned.